components/General/FaramGroup/FaramGroupApi.js

```diff
@@ -3,111 +3,147 @@ import FormElementApi from '../Form/FormElementApi';
-    );
+    constructor() {
+        super();
+        this.handlers = this.getHandler();
+    }
+
+    getHandler() {
+        return {
+            ...super.getHandler(),
+            input: {
+                getPropsFromApi: ({ faramElementName, faramInfo, ...otherProps }) => ({
+                    apiProps: faramElementName
+                        ? { faramElementName, faramInfo }
+                        : undefined,
+                    otherProps,
+                }),
+                calculateElementProps: ({ faramElementName, faramInfo }) => ({
+                    value: this.getValue(faramElementName),
+                    error: this.getError(faramElementName),
+                    onChange: this.getOnChange(faramElementName, faramInfo),
+                    disabled: this.isDisabled(),
+                    readOnly: this.isReadOnly(),
+                    changeDelay: this.getChangeDelay(),
+                }),
+            },
+            output: {
+                getPropsFromApi: ({ faramElementName, ...otherProps }) => ({
+                    apiProps: faramElementName
+                        ? { faramElementName }
+                        : undefined,
+                    otherProps,
+                }),
+                calculateElementProps: ({ faramElementName }) => ({
+                    value: this.getValue(faramElementName),
+                }),
+            },
+            errorMessage: {
+                getPropsFromApi: ({ faramElement, ...otherProps }) => ({
+                    apiProps: faramElement
+                        ? {}
+                        : undefined,
+                    otherProps,
+                }),
+                calculateElementProps: () => ({
+                    errors: this.getInternalError(),
+                }),
+            },
+            errorIndicator: {
+                getPropsFromApi: ({ faramElementName, ...otherProps }) => ({
+                    apiProps: faramElementName
+                        ? { faramElementName }
+                        : undefined,
+                    otherProps,
+                }),
+                calculateElementProps: ({ faramElementName }) => {
+                    const errors = this.getError(faramElementName);
+                    const calculatedProps = {
+                        hasError: analyzeErrors(errors),
+                        errors,
+                    };
+                    return calculatedProps;
+                },
+            },
+        };
+    }
 
-    getError = faramIdentifier => (this.props.error || {})[faramIdentifier];
+    // overrides FormElementApi
+    setProps = (props) => {
+        this.props = {
+            ...props,
+            onChange: (value, info) => {
+                // NOTE: Save these values in this.props so that above
+                // destructuring keeps working before setProps is
+                // again called.
+                this.value = value;
```

Comment:
Possible a bug, since `this.value` is not used anywhere. It should probably be `this.props.value`.